cpp/ps0005/cpp_ps0005.cpp

```diff
@@ -58,30 +30,34 @@ int main()
             break;
 
       case '/':
-            if(!num2) {
+            if(num1 && !num2) {
                   std::cout << "Math Error: cannot divide by 0." << std::endl;
-                  return 0;
+                  return 2;
+            }
+            if(!num1 && !num2) {
+                  std::cout << "Invalid character/s or zero valueInfinite number of solutions: (0 / 0) is Undefined Quantity!" << std::endl;
+                  return 3;
             }
             result = num1 / num2;
             break;
 
       case '%':
-            if(std::floor(num1) != num1 || std::floor(num2) != num2) {
+            if(std::int(num1) != num1 || std::int(num2) != num2) {
                   std::cout << "cannot execute modulo operation for floating point numbers." << std::endl;
-                  return 0;
+                  return 4;
             }
             result = static_cast<int>(num1) % static_cast<int>(num2);
             break;
       
       default:
             std::cout << "Invalid character for operator." << std::endl;
-            return 0;
-            break;
+            return 1;
       }
 
-      // printing the result if all data is valid
+      // printing the result if all data is valid 1/1
       std::cout << "(" << num1 << ") " << op << " (" << num2 << ")" << " = " << result << std::endl;
 
 
       return 0;
-}
\ No newline at end of file
+}
+// great work 😎 9/10 ⭐

```

Comment:
- [x] program is in the correct path
- [x] no syntax errors while compilation
- [x] formatted code to be clear and readable
- [ ] no bugs or warnings while execution
- [x] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case
+1 bonus mark for your effort ⭐

you can check the [problem solution](https://onlinegdb.com/2mF7i4R8H) for more